### DISCUSSION TITLE
How does the package omit empty space on bokeh plot?

### DISCUSSION THREAD

**Discussion by wolfcuring** [kernc/backtesting.py#435] [Q&A]
When plotting a timeseries, bokeh leaves empty space where there is no data, for example in the weekend. 
The regular solution is to have a major_label_overrides method. While this leaves further issue if you want to add another line, like a sma. 

I read kernc's code, but did not figure out how he handled this. The only thing I noticed is to set x_axis_type='linear'. But I don't think this is the whole logic. 

Any idea?

**Comment by kernc** [comment]
`x_axis_type='linear'` is it. The rest are tooltips and this formatter:
https://github.com/kernc/backtesting.py/blob/4f21a652d5292f93d5a717cc25f068fd2c35dcc2/backtesting/_plotting.py#L244-L255

**Reply by wolfcuring** [reply]
Thank you Kernc, I also noticed you used df.reset_index() to use numieric RangeIndex. So that is the same logic with major_label_overrides.
